File: code/src/main/java/service/ServiceUser.java
package service;

import model.User;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import repository.interfaces.RepositoryInterfaceUser;
import util.EncryptPassword;
import validator.Validator;

import java.util.Optional;


public class ServiceUser {
    private RepositoryInterfaceUser repository;
    private Validator<User> validator;
    private static final Logger logger = LogManager.getLogger();

    public ServiceUser(RepositoryInterfaceUser repository, Validator<User> validator) {
        logger.info("Initializing ServiceUser.");
        this.repository = repository;
        this.validator = validator;
    }

    public Optional<User> save(String username, String password) {
        logger.traceEntry("Saving user with username {} and password {}.", username, password);
        User user = new User(username, password);
        validator.validate(user);
        return repository.save(user);
    }

    public Optional<User> login(String username, String password) {
        logger.traceEntry("Login user with username {} and password {}.", username, password);
       // String encryptedPassword = EncryptPassword.encodePassword(password);
        return repository.checkConnection(username, password);
    }
}

File: MusicFestivalMPP/service/ServiceUser.cs
using System;
using log4net;
using MusicFestivalMPP.model;
using MusicFestivalMPP.repository.interfaces;
using MusicFestivalMPP.validator;

namespace MusicFestivalMPP.service;

public class ServiceUser
{
    private IRepositoryUser Repository;
    private IValidator<User> Validator;
    private static readonly ILog Log = LogManager.GetLogger("ServiceUser");

    public ServiceUser(IRepositoryUser repositoryUser, IValidator<User> validator)
    {
        Log.InfoFormat("Initializing ServiceUser.");
        Repository = repositoryUser;
        Validator = validator;
    }
    
    public User Save(String username, String password) {
        Log.InfoFormat("Saving user with username {0} and password {1}.", username, password);
        User user = new User(username, password);
        Validator.Validate(user);
        return Repository.Save(user);
    }

    public User Login(String username, String password) {
        Log.InfoFormat("Login user with username {0} and password {1}.", username, password);
        return Repository.CheckConnection(username, password);
    }
}